lib/Analyze.cpp

```diff
@@ -905,13 +936,41 @@ void RecoverMemoryAccesses(
     // TODO(pag): Leaving these as integers might be best, as we may go an
     //            collect them in the optimizer.
     } else {
-      LOG(ERROR) << "TODO: Found byte address " << std::hex << addr << std::dec
-                 << " that is mapped to memory but doesn't directly "
-                 << "resolve to a function or variable";
-
+      if (addr) {
+        LOG(ERROR) << "TODO: Found byte address " << std::hex << addr
+                   << std::dec
+                   << " that is mapped to memory but doesn't directly "
+                   << "resolve to a function or variable";
+        new_val = llvm::ConstantInt::get(int_type, addr, false);
+      } else {
+        new_val = used_val;
+      }
       new_val = llvm::ConstantInt::get(int_type, addr, false);
-    }
+      if (type) {
+
+        // 1) get the user instruction containing the use
+        llvm::User *test_user = use->getUser();
+        if (llvm::Instruction *inst =
+                llvm::dyn_cast<llvm::Instruction>(test_user);
+            inst) {
+          CHECK_EQ(inst->getOpcode(), llvm::Instruction::Add);
 
+          // 2) create an IRBuilder and // 3) set the insert point to just before the user instruction
```

Comment:
Add comment for `// 3)` to next line.

Reply:
Is there meant to be a `call` instruction added in here to replace the `add`?

Reply:
Whoops, so I forgot to delete this code. The location of the calls/adds is in the MCtoIRLifter instead. When we actually do the initial lifting we don't lift into variables, but instead into functions.